src/popup/popup.html

```diff
@@ -1 +1,19 @@
-<!-- popup.html -->
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="popup.css" />
+  </head>
+  <body>
+    <div class="container">
+      <p id="monitorStatus"></p>
+      <button id="startMonitorBtn">Start monitoring all extensions</button>
+      <button id="stopMonitorBtn">Stop monitoring all extensions</button>
+      <button id="actLogPage">View activity logs page</button>
+      <p id="errorText"></p>
+    </div>
+
+    <script src="popup.js" type="module"></script>
```

Comment:
In the test we are currently loading this HTML file to inject it into the JSDOM document and test the behaviors we expect, in that case we do not need (and we do not want) JSDOM to load this tag script because in the test we do create the Popup instance as part of the test.

To avoid loading the script tags when we do inject the content of the popup body as part of the test we could move the script tag into the head and add a `defer` attribute to it (so that it will basically behave as it does when the script tag is at the end of the body tag content).

Looking to the docs for the [`defer` attribute on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-defer) it looks that we don't even need to explicitly set the `defer` attribute of this `script` tag because it is `type="module` and it should be deferred automatically (which means that it would be executed after the document has been parsed, but before DOMContentLoad is fired, which is what we need given that popup.js is listening for that event and Popup class constructor does expect to be able to retrieve the DOM elements that are part of the document body).

Reply:
Thanks for explaining this.